### ISSUE TITLE
Support of HttpMethods PATCH/OPTIONS/HEAD

### ISSUE THREAD

**Issue by DigitalSmile** [teamcarma/swagger-jaxrs-doclet#49]
I have noticed that there is no support of this methods. Are there any plans to include them?

**Comment by conorroche** [comment]
should be straightforward to add, il aim to do it in the next few days

**Comment by conorroche** [comment]
i have now fixed this in the latest snapshot

**Comment by conorroche** [comment]
@DigitalSmile can you confirm this works as you expects now?

**Comment by DigitalSmile** [comment]
@conorroche I will return to you with results in next couple of days, thanks a lot!

**Comment by DigitalSmile** [comment]
@conorroche well, the 1.0.5-SNAPSHOT version gives me an exception in other unrelated staff I guess.
```bash
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.9:javadoc (generate-service-docs) on project web-server-layer: An error has occurred in JavaDocs report generation:
[ERROR] Exit code: 1 - javadoc: error - In doclet class com.carma.swagger.doclet.ServiceDoclet,  method start has thrown an exception java.lang.reflect.InvocationTargetException
[ERROR] java.lang.IllegalStateException: Min/Max values are not allowed for the type: string and format: null for the  field: message of the class: FeedbackDTO min value.
[ERROR] at com.carma.swagger.doclet.parser.ParserHelper.verifyNumericValue(ParserHelper.java:237)
[ERROR] at com.carma.swagger.doclet.parser.ApiModelParser.findReferencedElements(ApiModelParser.java:766)
[ERROR] at com.carma.swagger.doclet.parser.ApiModelParser.parseModel(ApiModelParser.java:194)
[ERROR] at com.carma.swagger.doclet.parser.ApiModelParser.parse(ApiModelParser.java:128)
[ERROR] at com.carma.swagger.doclet.parser.ApiMethodParser.generateParameters(ApiMethodParser.java:586)
[ERROR] at com.carma.swagger.doclet.parser.ApiMethodParser.parse(ApiMethodParser.java:175)
[ERROR] at com.carma.swagger.doclet.parser.CrossClassApiParser.parse(CrossClassApiParser.java:139)
[ERROR] at com.carma.swagger.doclet.parser.JaxRsAnnotationParser.run(JaxRsAnnotationParser.java:124)
[ERROR] at com.carma.swagger.doclet.ServiceDoclet.start(ServiceDoclet.java:23)
[ERROR] at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
[ERROR] at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] at java.lang.reflect.Method.invoke(Method.java:606)
[ERROR] at com.sun.tools.javadoc.DocletInvoker.invoke(DocletInvoker.java:280)
[ERROR] at com.sun.tools.javadoc.DocletInvoker.start(DocletInvoker.java:160)
[ERROR] at com.sun.tools.javadoc.Start.parseAndExecute(Start.java:397)
[ERROR] at com.sun.tools.javadoc.Start.begin(Start.java:167)
[ERROR] at com.sun.tools.javadoc.Main.execute(Main.java:59)
[ERROR] at com.sun.tools.javadoc.Main.main(Main.java:49)
[ERROR] 
[ERROR] Command line was: /opt/oracle-jdk-bin-1.7.0.65/jre/../bin/javadoc @options @packages
```
The FeedbackDTO is as follow:
```java
@XmlRootElement(name = "feedback")
public class FeedbackDTO {

    @Size(min = 0, max = USER_NAME_MAX_LENGTH)
    @JsonDeserialize(using = StringDeserializer.class)
    private String name;

    @Email(message = ValidationUtils.EMAIL_NOT_VALID)
    @Size(min = 0, max = ValidationUtils.EMAIL_MAX_LENGTH)
    @JsonDeserialize(using = StringDeserializer.class)
    private String email;

    @NotBlank(message = "Message is empty")
    @Size(min = ValidationUtils.MIN_LENGTH, max = ValidationUtils.FEEDBACK_MESSAGE_MAX_LENGTH)
    @JsonDeserialize(using = StringDeserializer.class)
    private String message;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}
```
We are using Hibernate Validation and I can say production 1.0.1 from Maven Central works normal. Any suggestions?

**Comment by DigitalSmile** [comment]
It seems this is about fix on javax.validation #51

**Comment by conorroche** [comment]
yes i will make a fix so that rather than validating that @size attributes are only on int/doubles that it allows size to be on Strings but ignores them for setting the max/min value

**Comment by conorroche** [comment]
@DigitalSmile ive committed a fix for this new issue which ive raised as issue #53

**Comment by DigitalSmile** [comment]
@conorroche Well done! Works as expected, thanks a lot. Any plans for the release in maven central?

**Comment by conorroche** [comment]
@DigitalSmile ive just released 1.0.4.1 as a patch of 1.0.4 which includes a fix for issue 53

**Comment by DigitalSmile** [comment]
@conorroche wonderful, thanks again!
